### PR TITLE
Upgrade Ember data 4.11.x to 4.12.x

### PR DESCRIPTION
**Pull request by Monkeychip**
Upgrading Ember data as a step closer to upgrading ember to 5.x.x

**TODO:**
- [x] Check if the OIDC provider changes are still necessary now with the ui-console test waiter.
- [ ] Smoke testing for basic workflows.
- [ ] Locally test pass after more than one run back to back (e.g. avoiding flakyness).
- [ ] Enterprise tests pass.

### CODE REVIEW

File: ui/tests/helpers/pki/pki-run-commands.js
Comment (Monkeychip) [inline ui/tests/helpers/pki/pki-run-commands.js:1]: ARG TODO look at making sure all the store clears are there.

File: ui/tests/helpers/pki/workflow.js
Comment (Monkeychip) [inline ui/tests/helpers/pki/workflow.js:1]: ARG TODO not being used.

File: ui/tests/helpers/pki/pki-run-commands.js [hunk 1]
```diff
@@ -16,9 +17,11 @@ export const tokenWithPolicy = async function (name, policy) {
   return consoleComponent.lastLogOutput;
 };
 
+// TODO: replace usage with runCmd
 export const runCommands = async function (commands) {
```
Comment (Monkeychip) [inline ui/tests/helpers/pki/pki-run-commands.js:21]: ARG TODO see if this is still being used by PKI

File: ui/package.json [hunk 2]
```diff
@@ -59,6 +59,7 @@
     "@babel/plugin-proposal-decorators": "^7.21.0",
     "@babel/plugin-proposal-object-rest-spread": "^7.12.1",
     "@babel/plugin-transform-block-scoping": "^7.12.1",
+    "@ember-data/legacy-compat": "^5.2.0",
```
Comment (hashishaw) [inline ui/package.json:62]: Found in embroider test branch -- this version should match our ember-data version 
```suggestion
    "@ember-data/legacy-compat": "~4.12.4",
```

**Pull request closed without merging**